### PR TITLE
Dashboard pages

### PR DESCRIPTION
**Pull request by Eibironke**
Has the home, steam, gog and epic dashboard pages

### CODE REVIEW

File: dashboard/pages/Epic.py [hunk 1]
```diff
@@ -1,18 +1,249 @@
-'''A script that creates the epic page for the dashboard.'''
+'''A script that creates the Steam page for the dashboard.'''
```
Comment (annalisev) [inline dashboard/pages/Epic.py:1]: should this say Epic?

File: dashboard/pages/GOG.py [hunk 2]
```diff
@@ -1,18 +1,255 @@
...
+    count_df = metrics_for_graphs_count(conn)
+    rating_df = metrics_for_graphs_rating(conn)
+
+    conn.close()
+
     st.set_page_config(page_title='GameScraper',
                        page_icon=":space_invader:", layout="wide")
 
-    st.title("GOG Summary")
+    st.title("Steam Summary")
```
Comment (annalisev) [inline dashboard/pages/GOG.py:233]: Should this be GOG?

File: dashboard/pages/Steam.py [hunk 3]
```diff
@@ -1,18 +1,256 @@
...
+                    IN {w_list}) 
+                    GROUP BY t.tag_name 
+                    LIMIT 10;""")
+        tag = cur.fetchall()
+
+    return pd.DataFrame(tag)
+
+
+def metrics_top_ten(conn_: connection) -> pd.DataFrame:
+    """Returns a Data-frame of top rated the games from the last week. """
```
Comment (annalisev) [inline dashboard/pages/Steam.py:133]:
```suggestion
    """Returns a Data-frame of top rated the games from the last week."""
```

**Review by annalisev — COMMENTED** [review]
Looks really good just maybe a few changes in the docstrings of some files?

File: dashboard/pages/Epic.py (hunk 1 above)
Comment (Eibironke) [reply dashboard/pages/Epic.py:1]: Great spot, my mistake

File: dashboard/pages/Epic.py [hunk 4]
```diff
@@ -1,18 +1,249 @@
...
+import altair as alt
 import streamlit as st
+from dotenv import load_dotenv
+
+from psycopg2 import connect
+from psycopg2.extras import RealDictCursor
+from psycopg2.extensions import connection
+
+
+def get_db_connection(config) -> connection:
```
Comment (shindym) [inline dashboard/pages/Epic.py:16]: could these functions be imported from another file? rather then repeating them again?

Comment (Eibironke) [reply dashboard/pages/Epic.py:16]: Absolutely can do for some but not all, will have a look 👍

File: dashboard/Home.py [hunk 5]
```diff
@@ -1,18 +1,248 @@
 '''A script that creates the home page for the dashboard.'''
 
 from os import environ as ENV
+from datetime import datetime, timedelta
 
+import pandas as pd
+import altair as alt
 import streamlit as st
+from dotenv import load_dotenv
+
```
Comment (hhm970) [inline dashboard/Home.py:10]: Get rid of blank line; they're all third-party libraries, and so should be in the same block

File: dashboard/Home.py [hunk 6]
```diff
@@ -1,18 +1,248 @@
...
+                    JOIN publisher as p
+                    on g.publisher_id = p.publisher_id
+                    WHERE g.release_date in {this_week_list}
+                    ORDER BY rating DESC LIMIT 20;""")
+        tags_ = cur.fetchall()
+
+    return pd.DataFrame(tags_)
+
+
+def price_chart(data_df: pd.DataFrame, sorting=True) -> alt.Chart:
```
Comment (hhm970) [inline dashboard/Home.py:150]:
```suggestion
def price_chart(data_df: pd.DataFrame, sorting: bool=True) -> alt.Chart:
```

File: dashboard/Home.py [hunk 7]
```diff
@@ -1,18 +1,248 @@
...
+    sort = "-y" if sorting else "x"
+
+    return alt.Chart(data_df).mark_bar().encode(
+        x=alt.Y("AVG price (£)", title='Average Daily Game Price'),
+        y=alt.X("release_date").sort(sort),
+        color="release_date"
+    )
+
+
+def count_chart(data_df: pd.DataFrame, sorting=True) -> alt.Chart:
```
Comment (hhm970) [inline dashboard/Home.py:164]:
```suggestion
def count_chart(data_df: pd.DataFrame, sorting:bool=True) -> alt.Chart:
```

File: dashboard/pages/Epic.py [hunk 8]
```diff
@@ -1,18 +1,249 @@
...
+
+    return pd.DataFrame(tag)
+
+
+def metrics_top_ten(conn_: connection) -> pd.DataFrame:
+    """Returns a Data-frame of top rated the games from the last week. """
+    w_list = get_week_list()
+
+    with conn_.cursor() as cur:
+        cur.execute(f"""SELECT g.name, g.rating, g.price, d.developer_name, p.publisher_name
```
Comment (hhm970) [inline dashboard/pages/Epic.py:138]: Nice SQL query

File: dashboard/pages/Epic.py [hunk 9]
```diff
@@ -1,18 +1,249 @@
...
+'''A script that creates the Epic games page for the dashboard.'''
 
 from os import environ as ENV
+from datetime import datetime, timedelta
 
+import pandas as pd
+import altair as alt
 import streamlit as st
+from dotenv import load_dotenv
+
```
Comment (hhm970) [inline dashboard/pages/Epic.py:10]:
```suggestion
```

File: dashboard/pages/Steam.py [hunk 10]
```diff
@@ -1,18 +1,256 @@
 '''A script that creates the Steam page for the dashboard.'''
 
 from os import environ as ENV
+from datetime import datetime, timedelta
 
+import pandas as pd
+import altair as alt
 import streamlit as st
+from dotenv import load_dotenv
+
```
Comment (hhm970) [inline dashboard/pages/Steam.py:10]:
```suggestion
```

File: dashboard/pages/GOG.py [hunk 11]
```diff
@@ -1,18 +1,255 @@
 '''A script that creates the GOG page for the dashboard.'''
 
 from os import environ as ENV
+from datetime import datetime, timedelta
 
+import pandas as pd
+import altair as alt
 import streamlit as st
+from dotenv import load_dotenv
+
```
Comment (hhm970) [inline dashboard/pages/GOG.py:10]:
```suggestion
```

**Review by hhm970 — COMMENTED** [review]
You've maintained an excellent consistency in your coding grammar throughout what is a thousand lines of code; well done!

I think it will be closer to being ready to merge after implementing changes

**Comment by Eibironke** [comment]
Made a lot of changes in the latest commit, hopefully should be a lot cleaner thanks to you guys 💯

**Review by annalisev — APPROVED** [review]
Looks good! I'm excited to see the dashboard!

**Pull request merged**